packages/core/src/external/quest/fhir/condition.ts

```diff
@@ -0,0 +1,94 @@
+import _ from "lodash";
+import { uuidv7 } from "@metriport/shared/util/uuid-v7";
+import { Condition, ConditionEvidence, Observation, Patient } from "@medplum/fhirtypes";
+import { ResponseDetail } from "../schema/response";
+import { ICD_10_URL, ICD_9_URL } from "@metriport/shared/medical";
+import { getPatientReference } from "./patient";
+import { getQuestDataSourceExtension } from "./shared";
+import { getObservationReference } from "./observation";
+import {
+  buildConditionVerificationStatus,
+  buildConditionClinicalStatus,
+} from "../../fhir/resources/condition";
+
+export function getConditions(
+  detail: ResponseDetail,
+  { patient, observation }: { patient: Patient; observation: Observation }
+): Condition[] {
+  const conditions = _([
+    getCondition({ patient, observation, diagnosisCode: detail.diagnosisCode1 }),
+    getCondition({ patient, observation, diagnosisCode: detail.diagnosisCode2 }),
+    getCondition({ patient, observation, diagnosisCode: detail.diagnosisCode3 }),
+    getCondition({ patient, observation, diagnosisCode: detail.diagnosisCode4 }),
+    getCondition({ patient, observation, diagnosisCode: detail.diagnosisCode5 }),
+    getCondition({ patient, observation, diagnosisCode: detail.diagnosisCode6 }),
+    getCondition({ patient, observation, diagnosisCode: detail.diagnosisCode7 }),
+    getCondition({ patient, observation, diagnosisCode: detail.diagnosisCode8 }),
+    getCondition({ patient, observation, diagnosisCode: detail.diagnosisCode9 }),
+    getCondition({ patient, observation, diagnosisCode: detail.diagnosisCode10 }),
+  ])
+    .compact()
+    .value();
+
+  return conditions;
+}
+
+function getCondition({
+  patient,
+  observation,
+  diagnosisCode,
+}: {
+  patient: Patient;
+  observation: Observation;
+  diagnosisCode?: string | undefined;
+}): Condition | undefined {
+  if (!diagnosisCode) return undefined;
+  const { system, code } = parseDiagnosisCode(diagnosisCode);
+  const subject = getPatientReference(patient);
+  const evidence = [getConditionEvidence(observation)];
+  const extension = [getQuestDataSourceExtension()];
+  const verificationStatus = buildConditionVerificationStatus("confirmed");
+  const clinicalStatus = buildConditionClinicalStatus("active");
+
+  return {
+    resourceType: "Condition",
+    id: uuidv7(),
+    subject,
+    evidence,
+    clinicalStatus,
+    verificationStatus,
+    code: {
+      coding: [
+        {
+          system,
+          code,
+        },
```

Comment:
@keshavsaharia is this going to get run through the term server to populate the display values for the conditions? Or does that come later in the pipeline downstream of the conversion bundle step for a given data source?

Reply:
Yes that is correct - this gets run through the term server after deduplication using `hydrateFhir`, which is returned in `hydratedBundle`.